Release Macaulay2 1.24.11

d-torrance opened the pull request:
Beginning work on the next release, planned for next Friday, Nov. 1.  I'll rebase this onto `development` as new PR's are accepted.

TODO:

- [ ] Certify latest packages published in JSAG
- [ ] Update ChangeLog
- [ ] Fix remaining issues in [version 1.24.11 milestone](https://github.com/Macaulay2/M2/milestone/18)
- [ ] Add docker recipe to build RHEL packages
- [ ] ???

mahrud commented on M2/Macaulay2/packages/Macaulay2Doc/changes.m2:
```diff
@@ -50,20 +50,38 @@ document {
+		LI { "The function ", TO changeDirectory, ", for changing the working directory, has been added." }
+		}
+	    },
+	LI { "changes to building Macaulay2:",
```
Is this a section that existed before? I don't think it's particularly relevant to users reading the html documentation ...

d-torrance replied:
No, this would be a new section. I wasn't sure if it was appropriate or not, and you've convinced me that it isn't. :)  I'll remove it on the next push

mahrud replied:
To be clear, there's plenty of things I would change about the change log ... maybe in a different format this would be very appropriate.

mahrud commented:
Do we have a specific cut off time for creating the release tag? Maybe midnight?

d-torrance commented:
My tentative plan was to create the tag in the morning (EST) on Friday so that there would be some built packages and documentation later in the day for an announcement.  But that may be wishful thinking!

d-torrance commented:
I've accounted for most of the user-facing changes since the last release in the changelog, with the exception of a few things I'm not as familiar with (complexes, basis, promotion, pseudocode).  I've tagged folks in the various PR's for help with these entries.

Suggestions for improvement would be much appreciated!

mahrud commented on M2/Macaulay2/editors/prism/package-lock.json:
Is this file necessary in git?

d-torrance replied:
Apparently, it's supposed to be: https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5

mahrud replied:
This package is essentially a glorified word list and file alone is bigger than it. Don't users just insert a couple of files in their HTML?

d-torrance replied:
Yeah, I was hesitant to add it but was blindly following the documentation. You've convinced me :) I'll remove it soon

mahrud reviewed (approved):
👍

mahrud commented:
Oops sorry that was by accident, although I guess it is ready?
